php/php.editor/src/org/netbeans/modules/php/editor/elements/MethodElementImpl.java

```diff
@@ -121,10 +126,66 @@ public static MethodElement createMagicMethod(final TypeElement type, String met
         return retval;
     }
 
+    public static MethodElement createMagicMethod(final TypeElement type, String methodName, int flags, List<Pair<String, String>> arguments, String returnType) {
+        MethodElement retval = new MethodElementImpl(
+                type,
+                methodName,
+                true,
+                0,
+                flags,
+                type.getFilenameUrl(),
+                null,
+                BaseFunctionElementSupport.ParametersImpl.create(fromParameterNames(arguments)),
+                BaseFunctionElementSupport.ReturnTypesImpl.create(returnType),
+                type.isDeprecated());
+        return retval;
+    }
+
+    static String getValidType(@NullAllowed String declaredType, @NullAllowed PhpVersion phpVersion) {
+        if (declaredType == null) {
+            return CodeUtils.EMPTY_STRING;
+        }
+        String type = declaredType.trim();
+        if (phpVersion != null) {
+            switch (type) {
+                case Type.MIXED:
+                    return phpVersion.hasMixedType() ? declaredType : CodeUtils.EMPTY_STRING;
+                case Type.OBJECT:
+                    return phpVersion.hasObjectType() ? declaredType : CodeUtils.EMPTY_STRING;
+                case Type.VOID:
+                    return phpVersion.hasVoidReturnType() ? declaredType : CodeUtils.EMPTY_STRING;
+                default:
+                    break;
+            }
+        }
+        return declaredType;
+    }
+
     private static List<ParameterElement> fromParameterNames(String... names) {
         List<ParameterElement> retval = new ArrayList<>();
         for (String parameterName : names) {
-            retval.add(new ParameterElementImpl(parameterName, null, 0, null, null, Collections.<TypeResolver>emptySet(), true, true, false, false, false, 0, false));
+            ParameterElement parameterElement = new ParameterElementImpl.Builder(parameterName)
+                    .isMagicMethod(false)
+                    .isMandatory(true)
+                    .isRawType(true)
+                    .build();
+            retval.add(parameterElement);
```

Comment:
Add the `Builder` because it's easy to understand what it is set.

Reply:
A good idea.